packages/truffle-deployer/test/helpers/reporter.js

```diff
@@ -0,0 +1,292 @@
+      confirmation: `${args[2]}`.padEnd(20) + '  > confirmation number:'.padEnd(25)  + args[0],
+      block:        `Block number: ${args[0]}  Wait: ${args[1]}`,
+    }
+
+    return kinds[kind];
+  }
+
+  async processDeploymentError(payload){
+    let message;
+
+    const error = payload.estimateError || payload.error;
+
+    const errors = {
+      INT: error.message.includes('base fee') || error.message.includes('intrinsic'),
+      OOG: error.message.includes('out of gas'),
+      RVT: error.message.includes('revert'),
+      ETH: error.message.includes('funds'),
+      BLK: error.message.includes('block gas limit'),
+      NCE: error.message.includes('nonce'),
+      INV: error.message.includes('invalid opcode'),
+    }
+
+    let type = Object.keys(errors).find(key => errors[key]);
+
+    switch (type) {
+      case 'INT':
+        (payload.gas)
+          ? message = this.messages('intWithGas', payload.contract, payload.gas, payload.blockLimit)
+          : message = this.messages('intNoGas', payload.contract, payload.blockLimit, payload.estimate);
+
+        this.deployer.logger.error(message);
+        break;
+
+      case 'OOG':
+        (payload.gas)
+          ? message = this.messages('intWithGas', payload.contract, payload.gas, payload.blockLimit)
+          : message = this.messages('oogNoGas', payload.contract, payload.blockLimit, payload.estimate);
+
+        this.deployer.logger.error(message);
+        break;
+
+      case 'RVT':
+        (payload.reason)
+          ? message = this.messages('rvtReason', payload.contract, payload.reason)
+          : message = this.messages('rvtNoReason', payload.contract);
+
+        this.deployer.logger.error(message);
+        break;
+
+      case 'INV':
+        (payload.reason)
+          ? message = this.messages('asrtReason', payload.contract, payload.reason)
+          : message = this.messages('asrtNoReason', payload.contract);
+
+        this.deployer.logger.error(message);
+        break;
+
+      case 'BLK':
+        (payload.gas)
+          ? message = this.messages('blockWithGas', payload.contract, payload.gas, payload.blockLimit)
+          : message = this.messages('blockNoGas', payload.contract, payload.blockLimit)
+
+        this.deployer.logger.error(message);
+        break;
+
+      case 'ETH':
+        let balance = await payload.contract.web3.eth.getBalance(payload.from);
+        balance = balance.toString();
+        message = this.messages('noMoney', payload.contract, payload.from, balance, error.message);
+        this.deployer.logger.error(message);
+        break;
+
+      case 'NCE':
+        message = this.messages('nonce', payload.contract, error.message);
+        this.deployer.logger.error(message);
+        break;
+
+      default:
+        message = this.messages('default', payload.contract, error.message);
+        this.deployer.logger.error(message);
+    }
+  }
+}
+
+module.exports = Reporter;
+
```

Comment:
I think we should pull this out, move the migrations reporter to a new modules (perhaps `truffle-reporters`) and have these test consume the production reporter.

Reply:
Definitely into this

Reply:
Ok cool I'll do that...